pkg/cable/vxlan/vxlan.go

```diff
@@ -0,0 +1,533 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package vxlan
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/log"
+	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/cable"
+	"github.com/submariner-io/submariner/pkg/natdiscovery"
+	"github.com/submariner-io/submariner/pkg/types"
+	"github.com/submariner-io/submariner/pkg/util"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"k8s.io/klog"
+)
+
+const (
+	VxLANIface             = "vxlan-tunnel"
```

Comment:
can we call it `submariner`  as for wireguard?

Reply:
Done.

Reply:
Changed the name back to vxlan-tunnel since cleanup logic which is no-ops now deleted the line created by wireguard. It may be a challenge when we support cable driver per connection.

Reply:
> Just one more nit - after that I'm good. @aswinsuryan There's quite a few commits with this PR - there some specific to review comments but I'm not sure if you want to squash all of them. Let me know if you want them all squashed on merge, otherwise please squash the ones you want when you push a final change.

I squashed those , most of them were not relevant as the cleanup driver was refactored.